AlphaPass.sol

```diff
@@ -40,12 +78,15 @@ contract AlphaNft is Ownable, ERC721A, ReentrancyGuard {
         }
         require(msg.value == cost, "wrong payment");
 
-
-        _isMinted[_msgSender()] = true;
         _safeMint(msg.sender, quantity);
+        increaseTokensMinted(quantity);
     }
 
     function mint(uint256 quantity) external payable nonReentrant {
+        require(_isPublicMintOpen, "public sales not opening");
+
+        require(_stage == 1 || _stage == 2 || _stage == 3, "invalid stage");
+        require(isStageMaxQtyExceed(quantity), "Exceed stage sales max limit");
```

Comment:
again, quantity checking not lock with address.
a address can mint as much as possible when `totalsupply + quantity <= maxQtyStageX`

Reply:
consider use
require(numberMinted(msg.sender) + quantity <= config.maxTokenPerAddress, "can not mint this many")
supported by 721a